app/models/katello/upstream_pool.rb

```diff
@@ -19,31 +24,52 @@ def response_to_pools(response)
       end
 
       def map_attributes(pool)
-        {
-          pool_id: pool['id'],
-          active: pool['activeSubscription'],
-          quantity: pool['quantity'],
-          start_date: pool['startDate'],
-          end_date: pool['endDate'],
-          contract_number: pool['contractNumber'],
-          consumed: pool['consumed'],
-          product_name: pool['productName'],
-          product_id: pool['productId'],
-          subscription_id: pool['subscriptionId']
+        attributes = {}
+        kat_to_cp_map = {
+          pool_id: 'id',
+          active: 'activeSubscription',
+          quantity: 'quantity',
+          start_date: 'startDate',
+          end_date: 'endDate',
+          contract_number: 'contractNumber',
+          consumed: 'consumed',
+          product_name: 'productName',
+          product_id: 'productId',
+          subscription_id: 'subscriptionId'
         }
+
+        kat_to_cp_map.map do |kat, cp|
+          attributes[kat] = pool[cp] if pool[cp]
+        end
```

Comment:
This isn't an improvement imo for two reasons:

1. Looping over a hash isn't performant (whereas accessing values by key *is*)
1. This code is longer and more confusing (I mean, it's not that confusing, but w/e)

I don't think either point is that strong; I'd like to know if you had strong reasons for changing it. Of course, if you make `kat_to_cp_map` it's own method, it might be worth it to keep this looping structure.

Reply:
The looping wasn't for performance or refactoring reasons, rather when we run w/ "quantities_only" and only id and quantity are returned from candlepin, all fields were still returned in the katello response and the rest of the fields were null. This was to add `if pool[cp]`. I'm refactoring based on @jturel's suggestions and I that should clean things up a bit